resources/Crop.php

```diff
@@ -0,0 +1,200 @@
+            $start = $start - $start;
+        }
+
+        // Correction for starting to late
+        if ($end > $image[$i]) {
+            $start = $start + $image[$i] - $end;
+            $end = $end + $image[$i] - $end;
+        }
+
+        // Return values for vertical crop
+        if ($p = 1) {
+            return [
+                'src_x' => 0,
+                'src_y' => $start,
+                'src_w' => $image[1],
+                'src_h' => $end - $start,
+            ];
+        }
+
+        // Return values for horizontal crop
+        return [
+            'src_x' => $start,
+            'src_y' => 0,
+            'src_w' => $end - $start,
+            'src_h' => $image[2]
+        ];
+
+    }
+
+    /**
+     * Return the src array of the attachment image containing url, width & height
+     *
+     * @param $imageId
+     * @return array|false
+     */
+    private function getImageSrc($imageId)
+    {
+        return wp_get_attachment_image_src($imageId, 'full');
+    }
+
+    /**
+     * Get all the image sizes excluding the ones that don't need cropping
+     *
+     * @return $this
+     */
+    public function getImageSizes()
+    {
+        global $_wp_additional_image_sizes;
+
+        // Get all the default WordPress image Sizes
+        foreach ((array)get_intermediate_image_sizes() as $imageSize) {
+            if (in_array($imageSize,
+                    ['thumbnail', 'medium', 'medium_large', 'large']) && get_option("{$imageSize}_crop")
+            ) {
+                $this->imageSizes[$imageSize] = [
+                    'width'  => get_option("{$imageSize}_size_w"),
+                    'height' => get_option("{$imageSize}_size_h"),
+                    'crop'   => (bool)get_option("{$imageSize}_crop"),
+                ];
+            }
+        }
+
+        // Get all the custom set image Sizes
+        foreach ($_wp_additional_image_sizes as $key => $imageSize) {
+            if ($imageSize['crop']) {
+                $this->imageSizes[$key] = $imageSize;
+            }
+        }
+
+        return $this;
+    }
+
+    /**
+     * Get the file destination based on the attachment in the argument
+     *
+     * @param $imageUri
+     * @param $imageSize
+     * @return mixed
+     */
+    private function getFilePath($imageUri, $imageSize)
+    {
+        // Get the path to the WordPress upload directory
+        $uploadDir = wp_upload_dir()['path'] . '/';
+
+        // Get the attachment name
+        $attachment = pathinfo($imageUri)['filename'];
+        $croppedAttachment = $attachment . '-' . $imageSize['width'] . 'x' . $imageSize['height'];
+
+        // Add the image size to the the name of the attachment
+        $fileName = str_replace($attachment, $croppedAttachment, basename($imageUri));
+
+        // Create the file path
+        $filePath = $uploadDir . $fileName;
+
+        // Remove the file to make sure we can upload the cropped image.
+        $this->removeOldFile($filePath);
```

Comment:
Should not happen at this position but in the function: `cropImage()` before calling `wp_crop_image()`